Run htmltest in CI to detect invalid links with hashes

DamianMaslanka5 opened the pull request:
Added checking if internal links have invalid hashes to avoid broken links, see https://github.com/ClickHouse/clickhouse-docs/issues/2398 and https://github.com/ClickHouse/clickhouse-docs/issues/2389

The links need to be fixed before merging this PR

Only links from [this repository are checked](https://github.com/DamianMaslanka5/ClickHouse/blob/c1da3f1e4190eb300e51f191908dfc7be0a31b1b/tests/ci/docs_check.py#L118) to avoid failing CI when broken links are introduced in https://github.com/ClickHouse/clickhouse-docs/

### Changelog category (leave one):
- Documentation (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ] <!---ci_set_required--> Allow: All Required Checks
- [ ] <!---ci_include_stateless--> Allow: Stateless tests
- [ ] <!---ci_include_stateful--> Allow: Stateful tests
- [ ] <!---ci_include_integration--> Allow: Integration Tests
- [ ] <!---ci_include_performance--> Allow: Performance tests
- [ ] <!---ci_set_normal_builds--> Allow: Normal Builds
- [ ] <!---ci_set_special_builds--> Allow: Special Builds
- [ ] <!---ci_set_non_required--> Allow: All NOT Required Checks
- [ ] <!---batch_0_1--> Allow: batch 1, 2 for multi-batch jobs
- [ ] <!---batch_2_3--> Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ] <!---ci_exclude_style--> Exclude: Style check
- [ ] <!---ci_exclude_fast--> Exclude: Fast test
- [ ] <!---ci_exclude_asan--> Exclude: All with ASAN
- [ ] <!---ci_exclude_tsan|msan|ubsan|coverage--> Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ] <!---ci_exclude_aarch64|release|debug--> Exclude: All with aarch64, release, debug
---
- [ ] <!---do_not_test--> Do not test
- [ ] <!---upload_all--> Upload binaries for special builds
- [ ] <!---no_merge_commit--> Disable merge-commit
- [ ] <!---no_ci_cache--> Disable CI cache

DamianMaslanka5 committed:
Run htmltest in CI to detect invalid links with hashes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=65248) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=65248) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ClickHouse/ClickHouse?pullRequest=65248) it.</sub>

robot-clickhouse-ci-2 commented:
<!-- automatic status comment for PR #65248 from DamianMaslanka5/ClickHouse:html-test-ci -->
*This is an automated comment for commit c1da3f1e4190eb300e51f191908dfc7be0a31b1b with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65248/c1da3f1e4190eb300e51f191908dfc7be0a31b1b/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

Felixoid committed:
Merge branch 'master' into html-test-ci

Felixoid committed:
Build htmltest in docs-builder

Felixoid committed:
Update a log file in TeePopen on write

Felixoid committed:
Multiple improvements of docs_check.py

- Run the script as a current user, not root
- Add `--no-pull-image` to use a local one
- Simplify logs analyze

Felixoid commented:
Hello @DamianMaslanka5, thanks for the contribution!

It's an exciting tool, and it finds a lot of broken URLs. You've done a great job trying to figure out our testing infrastructure! I slightly changed it to have a `htmltest` in our docs-builder image and changed the logic behind mounts and outputs. Let's see how it will run!

Felixoid commented:
Looks very promising, I am pressing the `Merge` button
